apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.Queries.cs

```diff
@@ -180,10 +184,14 @@ private JobsResource.InsertRequest CreateInsertQueryJobRequest(string sql, IEnum
             return CreateInsertJobRequest(new JobConfiguration { Query = jobConfigurationQuery, DryRun = options?.DryRun }, options);
         }
 
-        private TabledataResource.ListRequest CreateListRequest(TableReference tableReference, ListRowsOptions options)
+        private TabledataResource.ListRequest CreateListRequest(TableReference tableReference, ListRowsOptions options, TableSchema schema)
         {
             var request = Service.Tabledata.List(tableReference.ProjectId, tableReference.DatasetId, tableReference.TableId);
             options?.ModifyRequest(request);
+            if (schema != null)
```

Comment:
Possibly just use `request.SelectedFields = schema?.BuildSelectedFields();`? It's fine (I believe) to set `request.SelectedFields` to null redundantly.

Reply:
Or call `BuildSelectedFields` unconditionally, given that it looks like it will handle null input.